tests/src/Interop/Variant/VariantTest.cs

```diff
@@ -0,0 +1,360 @@
+    }
+
+    private unsafe static void TestOut()
+    {
+        Assert.IsTrue(Marshal_Out(out object obj));
+
+        Assert.IsTrue(obj is int);
+        Assert.AreEqual(NumericValue, (int)obj);
+    }
+
+    
+    private unsafe static void TestFieldByValue()
+    {
+        ObjectWrapper wrapper = new ObjectWrapper();
+
+        wrapper.value = (byte)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Byte));
+        
+        wrapper.value = (sbyte)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.SByte));
+        
+        wrapper.value = (short)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Int16));
+        
+        wrapper.value = (ushort)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.UInt16));
+        
+        wrapper.value = (int)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Int32));
+        
+        wrapper.value = (uint)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.UInt32));
+        
+        wrapper.value = (long)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Int64));
+        
+        wrapper.value = (ulong)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.UInt64));
+        
+        wrapper.value = (float)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Single));
+        
+        wrapper.value = (double)NumericValue; 
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Double));
+        
+        wrapper.value = StringValue;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.String));
+        
+        wrapper.value = CharValue;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Char));
+        
+        wrapper.value = true;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Boolean));
+        
+        wrapper.value = new DateTime(2018, 11, 6);
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.DateTime));
+        
+        wrapper.value = DecimalValue;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Decimal));
+        
+        wrapper.value = DBNull.Value;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Null));
+        
+        wrapper.value = System.Reflection.Missing.Value;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Missing));
+        
+        wrapper.value = null;
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Empty));
+        
+        wrapper.value = new object();
+        Assert.IsTrue(Marshal_Struct_ByValue(wrapper, ExpectedVariantType.Object));
+    }
+
```

Comment:
Please clean up your formatting here so the code is consistent with 0 noise. No extra lines, all properly aligned, et al.

Reply:
I actually prefer the extra whitespace between each of the small test cases when they are more than one line. Makes it easier to visually separate them.

Reply:
I understand having that delimiter and a single line is what is for that. if you would like additional space or a larger margin, I would suggest adding a comment instead of a blank line. There should be space for readability and clarity but space for the sake of space is confusing and interrupts the block structure of the class.

Reply:
Cleaned up the extra lines between the methods.